### PR TITLE
Added getFaces() method and stretch filter

### PR DESCRIPTION
**Pull request by chaithuzz2**
We have a face detection now for simplecv-js.
also i added a stretch filter which does the same function in SimpleCV

### CODE REVIEW

**Comment by chaithuzz2** [comment]
sorry for the inconvenience . deleted Image.coffee~

File: app/models/Image.coffee [hunk 1]
```diff
@@ -1161,7 +1161,174 @@ module.exports = class Image extends Model
...
+    dl.noFill()
+    dl.ellipse(x,y,width,height)
+    return @
+    
+  #Draws a circle with (x,y) as center with radius
+  drawCircle:(x,y,radius)=>
+    return @drawEllipse(x,y,radius,radius)
+    
+  #Draw a triangle with vertices (x1,y1) (x2,y2) (x3,y3)
+  drawTriangle:(x1,y1,x2,y2,x3,y3)=>
```
Comment (brooksryba) [inline app/models/Image.coffee]: Redundant, don't think we need functions like this in here

File: app/models/Image.coffee [hunk 2]
```diff
@@ -1161,7 +1161,174 @@ module.exports = class Image extends Model
...
+  
+  #Draws an ellipse centered at (x,y) with a height and width
+  drawEllipse:(x,y,width,height)=>
+    dl = @addDrawingLayer()
+    dl.noFill()
+    dl.ellipse(x,y,width,height)
+    return @
+    
+  #Draws a circle with (x,y) as center with radius
+  drawCircle:(x,y,radius)=>
```
Comment (brooksryba) [inline app/models/Image.coffee]: ""

File: app/models/Image.coffee [hunk 3]
```diff
@@ -1161,7 +1161,174 @@ module.exports = class Image extends Model
...
+    return @
+    
+  #Draws a line from (x1,y1) to (x2,y2) 
+  drawLine:(x1,y1,x2,y2)=>
+    dl = @addDrawingLayer()
+    dl.line(x1,y1,x2,y2)
+    return @
+  
+  #Draws an ellipse centered at (x,y) with a height and width
+  drawEllipse:(x,y,width,height)=>
```
Comment (brooksryba) [inline app/models/Image.coffee]: ""

File: app/models/Image.coffee [hunk 4]
```diff
@@ -1161,7 +1161,174 @@ module.exports = class Image extends Model
...
+  
+  #Draws a rectangular box in the image . the default drawing mode being from the corner.
+  drawRect:(x,y,width,height)=>
+    dl = @addDrawingLayer()
+    dl.noFill()
+    dl.rect(x,y,width,height)
+    return @
+    
+  #Draws a line from (x1,y1) to (x2,y2) 
+  drawLine:(x1,y1,x2,y2)=>
```
Comment (brooksryba) [inline app/models/Image.coffee]: ""

File: app/models/Image.coffee [hunk 5]
```diff
@@ -1161,7 +1161,174 @@ module.exports = class Image extends Model
...
+      if(gray[i] > high)
+        out.data[a] = 255; out.data[a+1] = 255; out.data[a+2] = 255
+      else
+        out.data[a] = out.data[a+1] = out.data[a+2] = gray[i]
+      i+=1
+      a+=4
+    return new Image(out)
+  
+  #Draws a rectangular box in the image . the default drawing mode being from the corner.
+  drawRect:(x,y,width,height)=>
```
Comment (brooksryba) [inline app/models/Image.coffee]: ""

File: app/models/Image.coffee [hunk 6]
```diff
@@ -1161,7 +1161,154 @@ module.exports = class Image extends Model
...
+    
+  #this is a basic face detection method which when used with image like this
+  #faces  = i.getFaces()
+  #gives a set of faces 
+  getFaces:()=>
+    dl  = @addDrawingLayer()
+    dl.noFill()
+    comp = ccv.detect_objects({"canvas" : (@canvas),"cascade" : cascade,"interval" : 5,"min_neighbors" : 1})
+    for face in comp
+      dl.rect(face.x,face.y,face.width,face.height)
```
Comment (brooksryba) [inline app/models/Image.coffee]: Do not draw in this function, do not force the user to do any markup.

**Comment by brooksryba** [comment]
Looks good, thanks!

**Pull request merged**